### PR TITLE
Add an example json for using toBeDetached flag for data disks

### PR DESCRIPTION
**Pull request by vamuzumd**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6257 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6257/commits/a3c306996fe3bcad5dbba0ad51de764c5601cd26

**Review by NelsonDaniel — CHANGES_REQUESTED** [review]
@vamuzumd Required checks are not passing because the example you added is not referenced anywhere. That needs to be fixed.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            "dataDisks": [
+              {
+                "diskSizeGB": 1023,
+                "createOption": "Empty",
+                "lun": 0,
+                "toBeDetached": true
+              },
+              {
+                "diskSizeGB": 1023,
+                "createOption": "Empty",
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json]: Pls. also specify the toBeDetached flag for the second data disk

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                }
+              }
+            ]
+          }
+        },
+        "name": "myVM"
+      }
+    },
+    "responses": {
+      "200": { }
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json]: The response payload contains the complete VM api model, pls. refer to CreateAVmWithADiffOsDisk.json file in the examples folder.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json]: This example file needs to be referenced in compute.json file in the examples section.

**Review by smotwani — CHANGES_REQUESTED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json (hunk 1 above)
Comment (vamuzumd) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json]: added

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json (hunk 3 above)
Comment (vamuzumd) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json]: added

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json (hunk 2 above)
Comment (vamuzumd) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/examples/DetachADataDiskUsingToBeDetachedFlag.json]: updated

**Comment by vamuzumd** [comment]
> @vamuzumd Required checks are not passing because the example you added is not referenced anywhere. That needs to be fixed.

Added a reference, please check the latest iteration.

**Review by NelsonDaniel — CHANGES_REQUESTED** [review]
@vamuzumd model validation is not passing. You can check the log here https://dev.azure.com/azure-sdk/public/_build/results?buildId=31222

File: specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json [hunk 4]
```diff
@@ -1886,7 +1886,13 @@
             }
           }
         },
-        "x-ms-long-running-operation": true
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Update a VM by detaching data disk": {
+            "$ref": "./examples/UpdateVMDetachDataDiskWithToBeDetachedFlag.json"
```
Comment (smotwani) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: Can we rename this file to UpdateVMDetachDataDiskWithToBeDetachedProperty.json

Comment (vamuzumd) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2019-03-01/compute.json]: done

**Comment by vamuzumd** [comment]
> @vamuzumd model validation is not passing. You can check the log here https://dev.azure.com/azure-sdk/public/_build/results?buildId=31222

I have fixed the validation issues

**Review by NelsonDaniel — APPROVED** [review]

**Comment by lmazuel** [comment]
Discussed with @NelsonDaniel error not related to PR, merging as asked.

**Pull request merged**